src/file_patcher.rs

```diff
@@ -14,13 +15,14 @@ pub struct FilePatcher {
 }
 
 impl FilePatcher {
-    pub fn new(path: &Path, query: &Query) -> Result<Option<FilePatcher>, std::io::Error> {
+    pub fn new(path: &Path, query: &Query) -> Result<Option<FilePatcher>> {
         let mut replacements = vec![];
-        let file = File::open(&path)?;
+        let file =
+            File::open(&path).with_context(|| format!("Could not open {}", path.display()))?;
         let reader = BufReader::new(file);
         let mut new_contents = String::new();
         for (num, chunk) in reader.split(b'\n').enumerate() {
```

Comment:
Maybe use `lines` instead of `split(b'\n')`?

Reply:
I need to keep line endings when writing the fil back later on. Otherwise if you have a CRLF file, ruplaced lines end with CR ...

Worth a comment though